scripts/build/Jenkins/Jenkinsfile

```diff
@@ -11,6 +11,7 @@ PIPELINE_CONFIG_FILE = 'scripts/build/Jenkins/lumberyard.json'
 INCREMENTAL_BUILD_SCRIPT_PATH = 'scripts/build/bootstrap/incremental_build_util.py'
 EBS_SNAPSHOT_SCRIPT_PATH = 'scripts/build/tools/ebs_snapshot.py'
 PIPELINE_RETRY_ATTEMPTS = 3
+GITHUB_URL = "https://github.com/aws-lumberyard-dev/o3de/blob/development/"
```

Comment:
Can we make this an env var instead. For example, for the o3de AR this should be set to `https://github.com/o3de/o3de`

Reply:
The branch name may also have to be set as a parameter, unless this is only meant for the development branch.

Reply:
Good catch. Totally missed that. 

Even as an env var, I don't think we should make this hardcoded to any particular repo or branch at all. If it starts with `github.com` this should be replaced. Optimally, it should be any https path, but unknown if we end up catching the wrong site if so.

Reply:
Agreed I think the repo URL used during checkout can be retrieved within the pipeline along with the branch name.

Reply:
So I just tested my changes and there's an issue with using the branch name because of PR runs. I'll get a path i.e.
`(https://github.com/o3de/o3de/tree/PR-8005/Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp#L420)`
which doesn't exist due to the PR branch. So either I always use development branch, or I default to development if it's a PR build. thoughts?

Reply:
Good catch, the formatting will be a bit different for PR builds. During checkout it pulls from the source branch using the commit id. 

If you replace the branch name with the commit id, that should work. For example to view the Jenkinsfile for this PR you would go to https://github.com/o3de/o3de/tree/15201ab77cbfe8109bcaa354459b1d33ebbcab11/scripts/build/Jenkins/Jenkinsfile

`15201ab77cbfe8109bcaa354459b1d33ebbcab11` Is the latest commit id in your PR. 

This also works for the development branch, so you can also apply it there. For example these URLs will point to the same file
- https://github.com/o3de/o3de/tree/development/scripts/build/Jenkins/Jenkinsfile
- https://github.com/o3de/o3de/tree/b94961dcfcf3b6ae880bf0557e6faf3d2734130d/scripts/build/Jenkins/Jenkinsfile (replace development with the latest commit id in that branch)

Reply:
Using the commit id is probably a better option anyway since these files can change between AR runs.